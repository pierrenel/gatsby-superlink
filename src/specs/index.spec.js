// @ts-nocheck It's drunk. Component props defined as optional are detected as required.
import React from "react"
import renderer from "react-test-renderer"
import Superlink from "../index"

global = Object.assign(global, { ___loader: { enqueue: jest.fn() } })

/**
 * @param {JSX.Element} component
 */
const getSnapshot = (component) => {
  return renderer.create(component).toJSON()
}

describe(`<Superlink />`, () => {
  describe(`internal links`, () => {
    it(`as default`, () => {
      const component = getSnapshot(
        <Superlink
          to="/blog"
        >
          Don't Panic!
      </Superlink>
      )
      expect(component).toMatchSnapshot()
    })

    it(`as full of props`, () => {
      const component = getSnapshot(
        <Superlink
          to="/blog"
          activeClassName="dolphin"
          partiallyActive={true}
          otherThing="otherThing"
        >
          So long and thanks for all the fish.
      </Superlink>
      )
      expect(component).toMatchSnapshot()
    })
  })

  describe(`external links`, () => {
    it(`as default`, () => {
      const component = getSnapshot(
        <Superlink
          to="https://douglas.adams"
        >
          Don't Panic!
      </Superlink>
      )
      expect(component).toMatchSnapshot()
    })

    it(`as full of props`, () => {
      const component = getSnapshot(
        <Superlink
          to="https://douglas.adams"
          ariaLabel="Douglas Adams"
          target="_blank"
        >
          Don't Panic!
      </Superlink>
      )
      expect(component).toMatchSnapshot()
    })
  })
})
