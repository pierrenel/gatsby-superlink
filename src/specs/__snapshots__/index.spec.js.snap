// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Superlink /> external links as default 1`] = `
<a
  href="https://douglas.adams"
  rel="nofollow noopener noreferrer"
>
  Don't Panic!
</a>
`;

exports[`<Superlink /> external links as full of props 1`] = `
<a
  ariaLabel="Douglas Adams"
  href="https://douglas.adams"
  rel="nofollow noopener noreferrer"
  target="_blank"
>
  Don't Panic!
</a>
`;

exports[`<Superlink /> internal links as default 1`] = `
<a
  href="/blog"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Don't Panic!
</a>
`;

exports[`<Superlink /> internal links as full of props 1`] = `
<a
  href="/blog"
  onClick={[Function]}
  onMouseEnter={[Function]}
  otherThing="otherThing"
>
  So long and thanks for all the fish.
</a>
`;
